### PR TITLE
Automate addition of core components to latest generated version

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Failed check: 

```
 Error: SA5001: should check returned error before deferring fsource.Close() (staticcheck)
```

**Comment by Revolyssup** [comment]
> Failed check:
> 
> ```
>  Error: SA5001: should check returned error before deferring fsource.Close() (staticcheck)
> ```

Yep, fixed it.

File: adapter/oam.go [hunk 1]
```diff
@@ -216,9 +223,78 @@ func CreateComponents(scfg StaticCompConfig) error {
...
+	if err != nil {
+		return ErrCreatingComponents(err)
+	}
 	return nil
 }
 
+//Currently versioning of Meshery's core components from a given service mesh adapter is corresponding
+//to the service mesh version itself. So everytime components are generated for a new service mesh version,
+// existing core components from previously latest version are to be copied in this new version.
+//Further we can change the schema of copied core components in latest version, or leave it as it is.
```
Comment (leecalcote) [inline adapter/oam.go:236]: "Core" components being like "Istio Mesh"?

Comment (Revolyssup) [reply adapter/oam.go:236]: Yes

File: adapter/oam.go [hunk 2]
```diff
@@ -216,9 +223,78 @@ func CreateComponents(scfg StaticCompConfig) error {
...
+	files, err := ioutil.ReadDir(fromDir)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		//core definition file or core schema file
+		if !strings.Contains(strings.TrimSuffix(f.Name(), ".json"), ".") || !strings.Contains(strings.TrimSuffix(f.Name(), ".meshery.layer5io.schema.json"), ".") {
+			fsource, err := os.Open(filepath.Join(fromDir, f.Name()))
+			if err != nil {
+				return err
```
Comment (leecalcote) [inline adapter/oam.go:247]: No new or existing error code to use here?

File: adapter/oam.go [hunk 3]
```diff
@@ -216,9 +223,78 @@ func CreateComponents(scfg StaticCompConfig) error {
...
+			//only for definition files
+			if !strings.Contains(strings.TrimSuffix(f.Name(), ".json"), ".") {
+				content, err = modifyVersionInDefinition(content, newVersion)
+				if err != nil {
+					return err
+				}
+			}
+			err = ioutil.WriteFile(filepath.Join(toDir, f.Name()), content, 0666)
+			if err != nil {
+				return err
```
Comment (leecalcote) [inline adapter/oam.go:263]: No new or existing error code to use here?

File: adapter/oam.go (hunk 2 above)
Comment (Revolyssup) [reply adapter/oam.go:247]: These are helper functions and these errors will be wrapped inside meshkit errors before being thrown.

File: adapter/oam.go (hunk 3 above)
Comment (Revolyssup) [reply adapter/oam.go:263]: These are helper functions and these errors will be wrapped inside meshkit errors before being thrown.

**Pull request merged**